components/calendar/src/date.rs

```diff
@@ -34,10 +34,21 @@ impl<C: Calendar> AsCalendar for Rc<C> {
     }
 }
 
-/// A date for a given calendar
+/// A date for a given calendar.
 ///
 /// This can work with wrappers around [`Calendar`] types,
-/// e.g. `Rc<C>`, via the [`AsCalendar`] trait
+/// e.g. `Rc<C>`, via the [`AsCalendar`] trait.
+///
+/// ```rust
+/// use icu::calendar::Date;
+///
+/// // Example: creation of ISO date from integers.
+/// let date_iso = Date::new_iso_date_from_integers(1970, 1, 2).unwrap();
```

Comment:
`nit`: please, use `expect` over `unwrap` - having `expect` everywhere in visible code/docs helps to understand what a failure means.

Reply:
Thank you for the feedback! For my understanding of Rust, are there cases where `unwrap` would be preferred?

Reply:
You can use it in integration tests and unit tests which are not exposed to readers. When i code I often use unwrap for convenience and then swap for expect when i clean up for review. It also gives me a chance to review if i can bubble up, and if a given fallibility is still needed (sometimes you'll add it at some point and then refactor removing the reason but keep the signature. It's a good chance to revisit)

Reply:
That makes sense -- thanks for the explanation!